app/controllers/application_controller.rb

```diff
@@ -24,6 +24,65 @@ def render_404
     render file: Rails.root.join("public", "404"), layout: false, status: "404"
   end
 
+    def put_and_get_search_params_in_session(controller_name, params_hash , filter)
+    session[:listing_filter_params]  ||= Hash.new
+
+    search_params_hash = {'search' => {}, 'page' => 1, 'per_page' => 10, 'reseted' => nil}
```

Comment:
Line is too long. [90/80]<br>Space inside { missing.<br>Space inside } missing.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.